packages/fast-usdc/src/exos/settler.js

```diff
@@ -227,6 +237,23 @@ export const prepareSettler = (
             statusManager.advanceOutcome(forwardingAddress, fullValue, success);
           }
         },
+        /**
+         * @param {ChainAddress} destination
+         *  @param {NobleAddress} forwardingAddress
+         * @param {Amount<'nat'>} fullAmount
+         * @param {EvmHash} txHash
+         * @throws {Error} if minted early, so advancer doesn't advance
+         */
+        checkMintedEarly(destination, forwardingAddress, fullAmount, txHash) {
+          const { value: fullValue } = fullAmount;
+          const key = makeMintedEarlyKey(forwardingAddress, fullValue);
+          const { mintedEarly } = this.state;
+          if (mintedEarly.has(key)) {
+            mintedEarly.delete(key);
+            void this.facets.self.forward(txHash, fullValue, destination.value);
+            throw Fail`${q(txHash)} minted before Observed`;
```

Comment:
This should be "raising a non-error" per [house style on Errors and Control Flow](https://github.com/Agoric/agoric-sdk/wiki/Errors-and-Control-Flow)